packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -118,18 +135,30 @@ export const prepareLiquidityPoolKit = (zone, zcf, brands, tools) => {
           /** @type {USDCProposalShapes['deposit']} */
           // @ts-expect-error ensured by proposalShape
           const proposal = lp.getProposal();
+          checkPoolBalance(poolSeat, shareWorth, USDC);
           const post = depositCalc(shareWorth, proposal);
+
           // COMMIT POINT
-          const mint = shareMint.mintGains(post.payouts);
-          zcf.atomicRearrange(
-            harden([
-              [lp, poolSeat, proposal.give],
-              [mint, lp, post.payouts],
-            ]),
-          );
-          lp.exit();
-          mint.exit();
-          await external.setShareWorth(post.shareWorth);
+
+          try {
+            const mint = shareMint.mintGains(post.payouts);
+            this.state.shareWorth = post.shareWorth;
+            zcf.atomicRearrange(
+              harden([
+                // zoe guarantees lp has proposal.give allocated
+                [lp, poolSeat, proposal.give],
+                // mintGains() above establishes that mint has post.payouts
+                [mint, lp, post.payouts],
+              ]),
+            );
+            lp.exit();
+            mint.exit();
+          } catch (bug) {
+            const reason = Error('🚨 cannot commit deposit', { cause: bug });
```

Comment:
`bug` is cute but a little distracting.
```suggestion
          } catch (cause) {
            const reason = Error('🚨 cannot commit deposit', { cause });
```

Reply:
oh. I didn't notice the cute part. I was trying to distinguish runtime problems from design-time bugs. but ok.